.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,50 @@
+name: GitHub Pages
+
+on:
+  push:
+    branches:
+    - 'master'
+
+jobs:
+  docs:
+    name: "Deploy"
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@master
+      - uses: webfactory/ssh-agent@v0.3.0
+        with:
+          ssh-private-key: ${{ secrets.SSH_PRIVATE_GH_PAGES }}
+      - name: Dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y xsltproc doxygen doxygen-doc texlive ghostscript graphviz
```

Comment:
Are all of these necessary?

Reply:
I think we only need `doxygen` and `graphviz`. Right now we are also generating the documentation in `latex` (i.e. check [here](https://github.com/robotology/osqp-eigen/tree/gh-pages/doxygen/doc/latex)). I think we can drop it.